### ISSUE TITLE
Function still cannot return empty list

### ISSUE THREAD

**Issue by xcharleslin** [hydro-project/cloudburst#15]
When I write a function that returns an empty list, it seems like the return value is never stored in the kvs, causing the .get() to hang forever. 

```
>>> res.get()
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/.local/lib/python3.6/site-packages/droplet/shared/future.py", line 26, in get
    obj = self.kvs_client.get(self.obj_id)[self.obj_id]
  File "/home/ubuntu/.local/lib/python3.6/site-packages/anna/client.py", line 106, in get
    KeyResponse)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/anna/zmq_util.py", line 27, in recv_response
    resp = rcv_sock.recv()
  File "zmq/backend/cython/socket.pyx", line 788, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 824, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
KeyboardInterrupt

>>> res.kvs_client.get(res.obj_id)
{'2e1d6152-5f96-4f18-a6e8-bfc60e79a0c5': None}
```

(This is distinct from the last version of this bug, where we were checking for "not obj" client-side instead of explicitly checking for None.)

**Comment by xcharleslin** [comment]
It seems we can't return dictionaries either, even if non-empty:

```
>>> def return_obj(droplet, obj):
...  return obj
... 
>>> cf = dc.register(return_obj, 'echo')
>>> res = cf(3)
>>> res.get()
3
>>> res = cf([3])
>>> res.get()
[3]
>>> res = cf([])
>>> res.get()
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/.local/lib/python3.6/site-packages/droplet/shared/future.py", line 26, in get
    obj = self.kvs_client.get(self.obj_id)[self.obj_id]
  File "/home/ubuntu/.local/lib/python3.6/site-packages/anna/client.py", line 106, in get
    KeyResponse)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/anna/zmq_util.py", line 27, in recv_response
    resp = rcv_sock.recv()
  File "zmq/backend/cython/socket.pyx", line 788, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 824, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
KeyboardInterrupt
>>> res = cf({3: 4})
>>> res.get()
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/.local/lib/python3.6/site-packages/droplet/shared/future.py", line 26, in get
    obj = self.kvs_client.get(self.obj_id)[self.obj_id]
KeyboardInterrupt
>>> res = cf('what')
>>> res.get()
'what'
>>> res = cf(dict())
>>> res.get()
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/.local/lib/python3.6/site-packages/droplet/shared/future.py", line 26, in get
    obj = self.kvs_client.get(self.obj_id)[self.obj_id]
  File "/home/ubuntu/.local/lib/python3.6/site-packages/anna/client.py", line 106, in get
    KeyResponse)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/anna/zmq_util.py", line 27, in recv_response
    resp = rcv_sock.recv()
  File "zmq/backend/cython/socket.pyx", line 788, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 824, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
KeyboardInterrupt
```

**Comment by vsreekanti** [comment]
Fixed by #17, hydro-project/anna#15.
